pkg/binding/bindingpolicy-resolver.go

```diff
@@ -402,8 +437,8 @@ func (resolver *bindingPolicyResolver) GetSingletonReportedStateRequestsForBindi
 	objIds := resolution.getWorkloadReferences()
 	ans := make([]SingletonReportedStateReturnStatus, len(objIds))
 	for idx, objId := range objIds {
-		want, qualifiedWECs := resolver.GetSingletonReportedStateRequestForObject(objId)
-		ans[idx] = SingletonReportedStateReturnStatus{objId, want, qualifiedWECs.Len()}
+		want, qualifiedWECsSingleton, _, _ := resolver.GetReportedStateRequestForObject(objId)
```

Comment:
Let's not compute a `sets.Set` and then discard it. Can be optimized in a follow-on PR.